src/tagmanager/tm_source_file.c

```diff
@@ -533,9 +539,17 @@ GPtrArray *tm_source_file_read_tags_file(const gchar *tags_file, TMParserType mo
 	}
 
 	file_tags = g_ptr_array_new();
-	while (NULL != (tag = new_tag_from_tags_file(NULL, fp, mode, format)))
-		g_ptr_array_add(file_tags, tag);
-	fclose(fp);
+	if (format == TM_FILE_FORMAT_CTAGS)
+	{
+		fclose(fp);  /* the readtags library opens the file by itself */
+		read_ctags_file(tags_file, mode, file_tags);
```

Comment:
Doesn't the `# format=ctags` confuse this parser?  I didn't try this yet, but it feels like it should:
* prevent pseudo tags (`!_…` ones) from being recognized as such
* get parsed as a normal entry

doesn't this happen?

Agreed this header it not necessary (and actually with supporting detection with some pseudo-tags probably not advisable), but it should either work, or we shouldn't pretend it still does.

Reply:
> doesn't this happen?

You are absolutely right. When `format=ctags` is found in the file, I added
```
g_warning("# format=ctags directive is no longer supported; please remove it from %s", tags_file);
```

I also updated the documentation because it advised to use `# format=ctags` (hopefully nobody reads it ;-)